web/src/auth.ts

```diff
@@ -15,13 +15,25 @@ export const { handlers, signIn, signOut, auth } = NextAuth({
     })
   ],
   callbacks: {
+    /**
+     * JWT Callback
+     * 
+     * This callback is called whenever a JSON Web Token is created (i.e. at sign in) 
+     * or updated (i.e whenever a session is accessed in the client). The returned value 
+     * will be encrypted, and it is stored in a cookie.
+     * 
+     * Requests to /api/auth/signin, /api/auth/session and calls to getSession(), 
+     * getServerSession(), useSession() will invoke this function, but only if you are 
+     * using a JWT session. This method is not invoked when you persist sessions in a database.
+     * 
+     * @see https://next-auth.js.org/configuration/callbacks
+     */
     async jwt({ token, account, profile }) {
       // Persist the OAuth access_token and or the user id to the token right after signin
       if (account && profile) {
         token.accessToken = account.access_token
```

Comment:
Please document that we are adding the access token from the provider.

```
GitHub’s access_token will give you access to GitHub’s APIs.
Self-managed providers (like [Keycloak](https://www.keycloak.org/), [oidc-provider](https://github.com/panva/node-oidc-provider), etc.) can be used to authorize against custom third-party backends.
```

Make it explict that we are checking github 

```
  if (account?.provider === "github") {
    return { ...token, accessToken: account.access_token }
  }
```